crates/runestick/src/vm_error.rs

```diff
@@ -176,6 +176,11 @@ pub enum VmErrorKind {
     Halted { halt: VmHaltInfo },
     #[error("failed to format argument")]
     FormatError,
+    #[error("standard format error: {error}")]
+    StdFormatError {
+        #[from]
+        error: std::fmt::Error,
+    },
```

Comment:
Look closely at the signature of `string_display_with`: `Result<fmt::Result, VmError>`. That means that it expects the `Ok` variant to be a `fmt::Result`, which is exactly what `write!` returns. The idea here is that formatting should propagate an error (which can be caught by user code) instead of a `VmError` (which cannot).

Reply:
Ah damn, stupid me....

Reply:
Not at all! It is a bit subtle.